### ISSUE TITLE
Installation error from devtools::install_github()

### ISSUE THREAD

**Issue by miepstei** [tidymodels/multilevelmod#15]
Hi,

thanks for all your great work on tidymodels. I came across this experimental package as I was looking for implementations for `parsnip `for mixed-effect models as a drop in replacement for `brms`. 

I'm trying to install the package in a docker image which can be launched as follows:

`docker run -it miepstei/tidymodels:0.1.1 R`

I then tried to install the package as suggested in the `README.md` but encountered an error.

The `reprex` is below but the actual error is not reported in the `reprex` output, but reported when I run the install outside of the reprex command and I included this after.

### reprex

``` r
devtools::install_github("tidymodels/multilevelmod",upgrade = FALSE)
#> Downloading GitHub repo tidymodels/multilevelmod@HEAD
#>      checking for file ‘/tmp/RtmpeUrGRv/remotes51455dddd75/tidymodels-multilevelmod-8d57d77/DESCRIPTION’ ...  ✓  checking for file ‘/tmp/RtmpeUrGRv/remotes51455dddd75/tidymodels-multilevelmod-8d57d77/DESCRIPTION’
#>   ─  preparing ‘multilevelmod’:
#>      checking DESCRIPTION meta-information ...  ✓  checking DESCRIPTION meta-information
#>   ─  checking for LF line-endings in source and make files and shell scripts
#>   ─  checking for empty or unneeded directories
#>   ─  building ‘multilevelmod_0.0.0.9000.tar.gz’
#>      
#> 
#> Installing package into '/usr/local/lib/R/site-library'
#> (as 'lib' is unspecified)
#> Warning in i.p(...): installation of package '/tmp/RtmpeUrGRv/file5142c062b9a/
#> multilevelmod_0.0.0.9000.tar.gz' had non-zero exit status
sessionInfo()
#> R version 4.1.0 (2021-05-18)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 20.04.3 LTS
#> 
#> Matrix products: default
#> BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
#> 
#> locale:
#>  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#>  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
#>  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] pillar_1.6.4      compiler_4.1.0    highr_0.9         remotes_2.4.1    
#>  [5] prettyunits_1.1.1 tools_4.1.0       testthat_3.1.0    digest_0.6.28    
#>  [9] pkgbuild_1.2.0    pkgload_1.2.3     evaluate_0.14     memoise_2.0.0    
#> [13] lifecycle_1.0.1   tibble_3.1.5      pkgconfig_2.0.3   rlang_0.4.12     
#> [17] reprex_2.0.1      cli_3.0.1         rstudioapi_0.13   curl_4.3.2       
#> [21] yaml_2.2.1        xfun_0.27         fastmap_1.1.0     withr_2.4.2      
#> [25] styler_1.5.1      stringr_1.4.0     knitr_1.36        desc_1.4.0       
#> [29] fs_1.5.0          vctrs_0.3.8       devtools_2.4.2    rprojroot_2.0.2  
#> [33] glue_1.4.2        R6_2.5.1          processx_3.5.2    fansi_0.5.0      
#> [37] rmarkdown_2.11    sessioninfo_1.1.1 purrr_0.3.4       callr_3.7.0      
#> [41] magrittr_2.0.1    usethis_2.1.0     backports_1.2.1   ps_1.6.0         
#> [45] ellipsis_0.3.2    htmltools_0.5.2   utf8_1.2.2        stringi_1.7.5    
#> [49] cachem_1.0.6      crayon_1.4.1
```

### Not included in reprex

``` r
Error: package or namespace load failed for ‘multilevelmod’:
 .onLoad failed in loadNamespace() for 'multilevelmod', details:
  call: NULL
  error: Model `poisson_reg` has not been registered.
Error: loading failed
Execution halted
```

**Comment by hfrick** [comment]
Hi! Could you try `devtools::install_github("tidymodels/multilevelmod")` without the `upgrade = FALSE`? multilevelmod needs the development version of parsnip.

@topepo Should multilevelmod require parsnip >= 0.1.7.9000 since that's the version which includes the model definitions of poisson_reg() etc?

**Comment by miepstei** [comment]
Hi,

thanks for your very quick reply. I tried that command but installing the development version of `parsnip` failed with

` Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
     namespace ‘generics’ 0.1.0 is already loaded, but >= 0.1.0.9000 is required
   Calls: <Anonymous> ... namespaceImportFrom -> asNamespace -> loadNamespace
   Execution halted
   ERROR: lazy loading failed for package ‘parsnip’
─  removing ‘/tmp/RtmpI62Qh2/Rinst2f04720df3/parsnip’`

I needed to install the development version of `generics` manually from github, and then run the command you stated above to get the package to install. This is probably because I install version `0.1.0` as a linux binary from `packagemanager.rstudio.com` specifically [from 2021-08-09](https://packagemanager.rstudio.com/cran/__linux__/focal/2021-08-09) and there hasn't been a compiled update to 0.1.1 for this date like there currently is on CRAN.

I thought initially this was a bug as in DESCRIPTION the development requirement for parsnip was not stated.

Thanks for your help.
